components/generator/imagevector/src/test/kotlin/io/github/composegears/valkyrie/generator/imagevector/TestUtils.kt

```diff
@@ -0,0 +1,18 @@
+package io.github.composegears.valkyrie.generator.imagevector
+
+import java.nio.file.Path
+import kotlin.io.path.readText
+import kotlin.io.path.toPath
+
+object TestUtils {
+
+    fun loadKotlinFile(name: String): String = getResourcePath(name).readText().uniteLine()
+
+    fun getResourcePath(name: String): Path {
+        val resource = TestUtils::class.java.classLoader.getResource(name) ?: error("Resource $name not found")
+
+        return resource.toURI().toPath()
+    }
+
+    private fun String.uniteLine(): String = replace("\r\n", "\n").replace("\r", "\n")
```

Comment:
```suggestion
    fun getResourceText(name: String): String = getResourcePath(name).readLines().joinToString("\n")

    fun getResourcePath(name: String): Path {
        val resource = TestUtils::class.java.classLoader.getResource(name) ?: error("Resource $name not found")

        return resource.toURI().toPath()
    }
```

Reply:
Add we can add one more thing:

```kt
fun OutputFormat.toResourceText(
    pathToBackingProperty: String,
    pathToLazyProperty: String,
): String {
    return when (this) {
        OutputFormat.BackingProperty -> TestUtils.getResourceText(pathToBackingProperty)
        OutputFormat.LazyDelegateProperty -> TestUtils.getResourceText(pathToLazyProperty)
    }.trim()
}
```

Reply:
With suggested

```kotlin 
fun getResourceText(name: String): String = getResourcePath(name).readLines().joinToString("\n")
```

tests failed due to missing empty line in expected file. looks like it ignores last empty line

<img width="1376" alt="image" src="https://github.com/user-attachments/assets/682e315c-73b9-45c2-be37-5463de0ec6f6">

can be
```kotlin
fun getResourceText(name: String): String = Files.readString(getResourcePath(name))
```
need to test on windows machine 🙂

Reply:
Yeah, this is why I added `trim` to address https://github.com/ComposeGears/Valkyrie/pull/70#discussion_r1696120522.

Reply:
Or you can add `.trimEnd()`.

Reply:
trim do nothing in this case (https://github.com/ComposeGears/Valkyrie/pull/70#discussion_r1696120522)
after loading resource it already miss end new line

trim necessary in this case for output generation 😏 
```kotlin
 val expected = outputFormat.toResourceText(
            pathToBackingProperty = "output/backing/WithoutPath.preview.kt",
            pathToLazyProperty = "output/lazy/WithoutPath.preview.kt",
        )
        assertThat(output.trim()).isEqualTo(expected)
```

Reply:
will push working solution, see how it works